### ISSUE TITLE
Error : TypeError: Cannot read properties of undefined (reading 'forEach')

### ISSUE THREAD

**Issue by jhso-dev** [callstack/repack#644]
Is it possible for **chunk** to be passed into **Platform**?
I encountered the following error and upon investigation, I found that **chunk** is being passed into **Platform**.
```
ℹ [00:44:56.585Z][WebpackCompilerWorker] env {
  bundleFilename: '',
  context: '/Users/x/dev/client/packages/app',
  reactNativePath: '/Users/x/dev/client/node_modules/react-native',
  mode: 'development',
  platform: 'chunk',
  devServer: { port: 8081, host: '', https: undefined, hmr: true }
}
✖ [00:44:56.603Z][DevServer] Failed to create code frame {
  reqId: 'req-e',
  error: "ENOENT: no such file or directory, open '/Users/x/dev/client/packages/app/node_modules/react-native/Libraries/LogBox/Data/LogBoxData.js'"
}
ℹ [00:44:56.604Z][DevServer] POST 200 /symbolicate request completed { responseTime: 12.083624988794327 }
platform chunk undefined
node:internal/event_target:1062
  process.nextTick(() => { throw err; });
                           ^

TypeError: Cannot read properties of undefined (reading 'forEach')
    at Worker.<anonymous> (/Users/x/dev/client/node_modules/@callstack/repack/dist/webpack/Compiler.js:107:36)
    at Worker.emit (node:events:517:28)
    at MessagePort.<anonymous> (node:internal/worker:256:53)
    at [nodejs.internal.kHybridDispatch] (node:internal/event_target:786:20)
    at exports.emitMessage (node:internal/per_context/messageport:23:28)

Node.js v18.20.2
```

**Comment by jbroma** [comment]
Hi @jhso-dev, what version of Re.Pack is this from?

**Comment by jhso-dev** [comment]
@jbroma Hi jbroma, I am using repack 3.7.0 and react-native 0.73.7. 
I also tried it with 4.1.1
Thanks in advance!

**Comment by jbroma** [comment]
Can you share a reproduction repository please? I'm not able to pinpoint anything from the trace you've provided. No clue how 'chunk' could end up as a platform 🤔. 

I also can't locate the error from this trace, I've check 3.7.0 and 4.1.1 and none of them match this:

```sh
TypeError: Cannot read properties of undefined (reading 'forEach')
    at Worker.<anonymous> (/Users/x/dev/client/node_modules/@callstack/repack/dist/webpack/Compiler.js:107:36)
    at Worker.emit (node:events:517:28)
    at MessagePort.<anonymous> (node:internal/worker:256:53)
    at [nodejs.internal.kHybridDispatch] (node:internal/event_target:786:20)
    at exports.emitMessage (node:internal/per_context/messageport:23:28)
```

**Comment by jhso-dev** [comment]
@jbroma Hi jbroma :) 

I did some research and found that the platform is parsed from the URL using getSourceMap, resulting in a chunk like this:
http://localhost:8081/ios/src_app2_tsx-node_modules_react-native-reanimated_src_reanimated2_sync_recursive.chunk.bundle

Please refer to the source below:
https://github.com/callstack/repack/blob/c0f9b7eaddffed993f488594d9ffd507214f9df9/packages/repack/src/commands/start.ts#L263

So, if I manually change the platform from parseFileUrl to either ios or android, it seems to work correctly.

**Comment by jbroma** [comment]
Oh, now with the added context it makes way more sense! Great find, that's definitely a bug that needs to be fixed, would you like to open a PR with a fix for that?

**Comment by jhso-dev** [comment]
@jbroma please review the pull request when you have a moment? Thank you!
https://github.com/callstack/repack/pull/647

**Comment by jbroma** [comment]
Fixed by #647
